superset/utils/decorators.py

```diff
@@ -72,6 +80,12 @@ def wrapper(*args: Any, **kwargs: Any) -> ETagResponseMixin:
             if request.method == "POST":
                 return f(*args, **kwargs)
 
+            # if it is dashboard request but feature is not eabled,
+            # do not use cache
+            is_dashboard = is_dashboard_request(kwargs)
```

Comment:
Instead of adding a helper function, maybe we can just expand `dashboard_id_or_slug` in `wrapper` and make this code more transparent?

```python
def wrapper(*args: Any, dashboard_id_or_slug: str=None, **kwargs: Any) -> ETagResponseMixin:
  # ...
  is_dashboard = dashboard_id_or_slug is not None
``` 

Better yet, we should probably aim for keeping all dashboard specific logics out of `etag_cache` so it stays generic. Maybe add a `skip=` parameter that runs the feature flag check.

```python
@etag_cache(skip=lambda: is_feature_enabled("ENABLE_DASHBOARD_ETAG_HEADER"))
```

```python
def etag_cache(
    max_age: int,
    check_perms: Callable[..., Any],
    skip: Optional[Callable[..., Any]] = None,
) -> Callable[..., Any]:

  def decorator(f: Callable[..., Any]) -> Callable[..., Any]:

    def wrapper(*args: Any, **kwargs: Any) -> ETagResponseMixin:

      check_perms(*args, **kwargs)

      if request.method == "POST" or (skip and skip(*args, **kwargs)):
        return f(*args, **kwargs)
```

Reply:
thanks! after introduce this `skip`, `dashboard_id` is not needed in decorator at all.